app/Resources/views/base.html.twig

```diff
@@ -83,7 +83,7 @@
                                     <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false"><i class="fa fa-globe"></i> <span class="caret"></span></a>
                                     <ul class="dropdown-menu locales" role="menu">
                                         {% for locale in locales() %}
-                                            <li {% if app.request.locale == locale.code %}class="active"{% endif %}><a href="{{ path('blog_index', { _locale: locale.code }) }}">{{ locale.name|capitalize }}</a></li>
+                                            <li {% if app.request.locale == locale.code %}class="active"{% endif %}><a href="{{ path(app.request.get('_route'), app.request.get('_route_params')|merge({ _locale: locale.code })) }}">{{ locale.name|capitalize }}</a></li>
```

Comment:
you should pass an empty array as default value for the route_params, to get an empty array rather than null when the error happens without route